Control.FullScreen.js

```diff
@@ -240,10 +240,10 @@
 			this._map._isFullscreen ? L.DomUtil.removeClass(this.link, 'leaflet-fullscreen-on') : L.DomUtil.addClass(this.link, 'leaflet-fullscreen-on');
 		},
 
-		_handleFullscreenChange: function () {
+		_handleFullscreenChange: function (ev) {
 			var map = this._map;
 			map.invalidateSize();
-			if (!this._screenfull.isFullscreen && !map._exitFired) {
+			if (ev.target === map.getContainer() && !this._screenfull.isFullscreen && !map._exitFired) {
 				map.fire('exitFullscreen');
 				map._exitFired = true;
 				map._isFullscreen = false;
```

Comment:
I think, this is the line after which `map.invalidateSize();` should be added (see [comment ](https://github.com/brunob/leaflet.fullscreen/pull/118#issuecomment-1881796644)from brunob)